src/gt4py/backend/gtc_backend/defir_to_gtir.py

```diff
@@ -53,6 +53,21 @@
 from gtc.common import ExprKind
 
 
+def _make_literal(v: numbers.Number) -> gtir.Literal:
+    value: Union[BuiltinLiteral, str]
+    if issubclass(type(v), bool):
+        dtype = common.DataType.BOOL
+        value = common.BuiltInLiteral.TRUE if v else common.BuiltInLiteral.FALSE
+    else:
+        dtype = (
+            common.DataType.INT64
+            if (cast(float, v) / 1.0).is_integer()
+            else common.DataType.FLOAT64
+        )
```

Comment:
I'm not convinced that this is a task that should be carried out in the upper part of the toolchain. Additionally what happens if you get neither float nor int? I would have expected a dict from python dtype to common.DataType here.

Reply:
This will be removed once the frontend writes to gtir directly. The `common.DataType` enumeration doesn't support anything else besides boolean, integer, and floating point types. If you can think of a way to check that I'd like to put it in.

Reply:
Not sure I understand the second sentence. What else could one need? I would propose something along these lines:
```
if isinstance(v, int):
  # note: could be extended to support 32 bit integers by checking the values magnitude
  dtype = common.DataType.INT64
elif isinstance(v, float):
  dtype = common.DataType.FLOAT64
else:
  print(f"Warning: Only INT64 and FLOAT64 literals are supported currently. Implicitly upcasting `{v}` to FLOAT64")
  dtype = common.DataType.FLOAT64
```